consensus/propagation/types/types_test.go

```diff
@@ -674,8 +694,7 @@ func TestHaveParts_ValidatePartHashes(t *testing.T) {
 			haveParts: HaveParts{
 				Parts: nil,
```

Comment:
this adds a constraint on haves to always have parts. Otherwise, will be rejected